geode-core/src/main/java/org/apache/geode/SystemFailure.java

```diff
@@ -376,10 +372,7 @@ protected static void runWatchDog() {
       logWarning(WATCHDOG_NAME, "Unable to initialize watchdog", t);
       return;
     }
-    for (;;) {
-      if (stopping) {
-        return;
-      }
+    while (!stopping) {
```

Comment:
according to Effective Java 3rd Edition item 78, the runtime JVM may optimize `while (!stopping) {}` into `if(!stopping) while(true) {}`.  It doesn't say whether the `volatile` keyword is sufficient to prevent this, but says synchronization is the only correct way.  there's also  quite a mixture in this file of when `stopping` is used within a synchronized block and when it isn't.  While you aren't expected to fix these pre-existing problems, be very cautious of subtle behavior changes that can result from innocent-looking code cleanup.  Might be safer just to leave it the way it was.

Reply:
Good catch @onichols-pivotal, but as I read this section `volatile` is sufficient, and is given as an example of a non-locking solution, to prevent the hoisting issue. Since `stopping` is declared volatile we shouldn't have a thread view issue here or a hoisting optimization.

Reply:
cool, volatile should be sufficient then